core/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java

```diff
@@ -182,7 +182,7 @@ public TypeParser() {
         protected FormatDateTimeFormatter dateTimeFormatter;
         protected DateMathParser dateMathParser;
 
-        DateFieldType() {
+        public DateFieldType() {
```

Comment:
I prefer to keep this package protected. I think in `CompositeAggregatorTests#setUp()` we should do this instead:

```java
DateFieldMapper.Builder builder = new DateFieldMapper.Builder("date");
builder.docValues(true);
DateFieldMapper fieldMapper = builder.build(new Mapper.BuilderContext(createIndexSettings().getSettings(), new ContentPath(0)));
FIELD_TYPES[3] = fieldMapper.fieldType();
```